### PR TITLE
Fix Bar Animation with the given Magic Number

### PR DESCRIPTION
**Pull request by huei-tan-sociomantic**
Before the Magic Number `width * 2`
recharts-bar-animation-w o-magic

After the Magic Number
recahrts-bar-animation-magic

similar to https://github.com/recharts/recharts/pull/1025

### CODE REVIEW

**Comment by xile611** [comment]
@huei-tan-sociomantic Thank you for all your answers and PRS. I'm really appreciate your contribution.  Maybe you can send me an email to exchange our thoughts about Recharts.

File: src/cartesian/Bar.js [hunk 1]
```diff
@@ -262,10 +263,16 @@ class Bar extends Component {
               }
 
               if (layout === 'horizontal') {
-                const interpolator = interpolateNumber(0, entry.height);
-                const h = interpolator(t);
+                const interpolatorX = interpolateNumber(width * 2, entry.x);
```
Comment (xile611) [inline src/cartesian/Bar.js]: @huei-tan-sociomantic Maybe you can add some comments here.

Comment (huei-tan-sociomantic) [reply src/cartesian/Bar.js]: DONE.

**Comment by huei-tan-sociomantic** [comment]
@xile611 what's your account name in Rocket.Chat Recharts channel?

**Comment by xile611** [comment]
@huei-tan-sociomantic You can send me an email,  dingling112@gmail.com.

And the initial animation of bar seems strange now ...

**Comment by DeviousM** [comment]
@xile611 - initial animation is totally broken. That's because of the `interpolatorX` usage as it's animating the X axis even in the initial render. IMO that PR should be reverted as it breaks more things than it brings to the table.

**Comment by huei-tan-sociomantic** [comment]
my bad, please revert this pull request 🤕

**Comment by huei-tan-sociomantic** [comment]
@xile611 I didn't receive your reply email since last year.

**Pull request merged**